framework/test/jest/unit/specs/modules/chain/blocks/blocks.spec.js

```diff
@@ -1703,12 +1704,113 @@ describe('blocks', () => {
 		});
 	});
 
-	describe('loadBlocksDataWs', () => {
-		it.todo('a');
-	});
+	describe('loadBlocksFromLastBlockId', () => {
+		describe('when called without lastBlockId', () => {
+			it('should reject with error', async () => {
+				expect.assertions(1);
+				try {
+					await blocksInstance.loadBlocksFromLastBlockId();
+				} catch (err) {
+					expect(err.message).toBe('lastBlockId needs to be specified');
+				}
+			});
+		});
+
+		describe('when called without limit', () => {
+			const validLastBlock = {
+				height: 100,
+				id: 'block-id',
+			};
+
+			beforeEach(async () => {
+				stubs.dependencies.storage.entities.Block.get.mockResolvedValue([
+					validLastBlock,
+				]);
+			});
+
+			it('should use limit 1 as default', async () => {
+				await blocksInstance.loadBlocksFromLastBlockId('block-id');
+				expect(
+					stubs.dependencies.storage.entities.Block.get,
+				).toHaveBeenCalledWith(
+					{
+						height_gt: 100,
+						height_lt: 101,
+					},
+					{
+						extended: true,
+						limit: null,
+						sort: ['height'],
+					},
+				);
+			});
+		});
+
+		describe('when called with invalid lastBlockId', () => {
+			beforeEach(async () => {
```

Comment:
No need for `beforeEach`, would prefer to add this setup directly to the test under `// Arrange` section.

Reply:
I think when we add the similar test case, this would be easier and ensure that stub is reset

Reply:
It's personal taste! If it's a small stub, I prefer it to be inside the test itself, but the test section below this one is a large stub which makes sense to put in a beforeEach block. For me it's fine to keep this kind of consistency. I'll resolve the comment